src/server/common/dbconn.cpp

```diff
@@ -147,31 +184,91 @@ std::unique_ptr<simple_wml::document> dbconn::get_game_history(int player_id, in
 "    select 1 "
 "    from "+db_game_player_info_table_+" player1 "
 "    where game.INSTANCE_UUID = player1.INSTANCE_UUID "
-"      and game.GAME_ID = player1.GAME_ID "
-"      and player1.USER_ID = ? "
-"  ) "
+"      and game.GAME_ID = player1.GAME_ID ";
+	game_history_query += player_id != 0 ? " and player1.USER_ID = ? " : " and player1.USER_ID != ? ";
+	game_history_query += "  ) "
 "  and game.INSTANCE_UUID = player.INSTANCE_UUID "
 "  and game.GAME_ID = player.GAME_ID "
 "  and player.USER_ID != -1 "
-"  and game.END_TIME is not NULL "
-"inner join "+db_game_content_info_table_+" scenario "
+"  and game.END_TIME is not NULL ";
+
+	if(!search_game_name.empty())
+	{
+		game_history_query += " and game.GAME_NAME like concat(";
+		game_history_query += search_game_name_leading ? "'%'," : "";
+		game_history_query += " ? ";
+		game_history_query += search_game_name_trailing ? ",'%')" : ")";
+	}
+	else
+	{
+		game_history_query += " and game.GAME_NAME != ? ";
+	}
+
+	game_history_query += "inner join "+db_game_content_info_table_+" scenario "
 "   on scenario.TYPE = 'scenario' "
 "  and scenario.INSTANCE_UUID = game.INSTANCE_UUID "
-"  and scenario.GAME_ID = game.GAME_ID "
-"inner join "+db_game_content_info_table_+" era "
+"  and scenario.GAME_ID = game.GAME_ID ";
+	if(search_content_type == 0)
+	{
+		if(!search_content.empty())
+		{
+			game_history_query += " and scenario.ID like concat(";
+			game_history_query += search_content_leading ? "'%'," : "";
+			game_history_query += " ? ";
+			game_history_query += search_content_trailing ? ",'%')" : ")";
```

Comment:
Is concat needed? Why not just replace `*` with `%` and `?` with `_` in the search strings?

Reply:
I tried that initially, and what ends up happening is the `%` gets treated as a literal part of the string rather than as a wildcard.

Reply:
It's certainly possible - I'll take a closer look.

Reply:
Updated to use `utils::to_sql_wildcards`.